File: algorithms/1_easy/F/03158_find_the_xor_of_numbers_which_appear_twice/solution.py
class Solution:
    def duplicateNumbersXOR(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        res = 0

        for i in range(1, n):
            if nums[i] == nums[i - 1]:
                res ^= nums[i]

        return res

File: algorithms/1_easy/F/03158_find_the_xor_of_numbers_which_appear_twice/solution.cpp
class Solution {
public:
    int duplicateNumbersXOR(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        const size_t len{nums.size()};
        int res{0};

        for (size_t i{1}; i < len; ++i) {
            if (nums[i] == nums[i - 1]) {
                res ^= nums[i];
            }
        }

        return res;
    }
};
